zcash_primitives/src/transaction/components/tze/builder.rs

```diff
@@ -32,13 +32,27 @@ impl fmt::Display for Error {
 
 #[allow(clippy::type_complexity)]
 pub struct TzeSigner<'a, BuildCtx> {
-    prevout: TzeOut,
     builder: Box<dyn FnOnce(&BuildCtx) -> Result<(u32, Vec<u8>), Error> + 'a>,
 }
 
+#[derive(Clone)]
+pub struct TzeBuildInput {
```

Comment:
Does this need to be public? AFAICT it is only used as a type inside this module, and otherwise is exposed as `impl TzeBuildInput` which does not require this type to be public.
```suggestion
struct TzeBuildInput {
```